classes/class-plugin.php

```diff
@@ -73,14 +73,24 @@ public function __construct() {
 		require_once $this->locations['inc_dir'] . 'functions.php';
 
 		// Load DB helper interface/class
-		$driver = '\WP_Stream\DB';
-		if ( class_exists( $driver ) ) {
-			$this->db = new DB( $this );
+		$driver_class = apply_filters( 'wp_stream_db_driver', '\WP_Stream\DB_Driver_WPDB' );
+		$driver       = null;
+
+		if ( class_exists( $driver_class ) ) {
+			$driver   = new $driver_class();
+			$this->db = new DB( $driver );
 		}
 
+		$error = false;
 		if ( ! $this->db ) {
+			$error = esc_html__( 'Stream: Could not load chosen DB driver.', 'stream' );
+		} elseif ( ! $driver instanceof DB_Driver ) {
+			$error = esc_html__( 'Stream: DB driver must implement DB Driver interface.', 'stream' );
+		}
+
+		if ( $error ) {
 			wp_die(
-				esc_html__( 'Stream: Could not load chosen DB driver.', 'stream' ),
+				esc_html( $error ),
```

Comment:
Can't use a variable as the first arg of `esc_html__()`.

Instead:
```
		if ( ! $this->db ) {
			$error = __( 'Stream: Could not load chosen DB driver.', 'stream' );
		} elseif ( ! $driver instanceof DB_Driver ) {
			$error = __( 'Stream: DB driver must implement DB Driver interface.', 'stream' );
		}
```

and then just:

```
				esc_html( $error ),
```